internal/services/containers/container_group_resource.go

```diff
@@ -1997,3 +1944,30 @@ func expandContainerGroupDnsConfig(input interface{}) *containerinstance.DnsConf
 
 	return nil
 }
+
+func flattenContainerGroupSubnets(input *[]containerinstance.ContainerGroupSubnetId) []interface{} {
+	subnetIDs := make([]interface{}, 0)
+	if input == nil {
+		return subnetIDs
+	}
+
+	for _, resourceRef := range *input {
+		if resourceRef.Id == "" {
+			continue
+		}
+
```

Comment:
The subnet ID wants to be parsed and then normalized (i.e. call the id.ID()) before appended to the `subnetIDs`.